NetCoreRestApi/DataLayer.EF/ProductMockDbContext.cs

```diff
@@ -1,78 +1,86 @@
-﻿using DataLayer.EF.Converters;
-using DataLayer.EF.Entities;
-using DataLayer.Models;
+﻿using DataLayer.EF.Entities;
 using Microsoft.EntityFrameworkCore;
 using System.Collections.Generic;
 
 namespace DataLayer.EF
 {
     public class ProductMockDbContext : ProductsDbContext
     {
-        public ProductMockDbContext(DbContextOptions<ProductsDbContext> options) : base(options)
+        private List<ProductEntity> _productModels;
+        private List<CategoryEntity> _categoryEntities;
+
+        public ProductMockDbContext(DbContextOptions<ProductMockDbContext> options) : base(options)
         {
-            AddRange(GetProductsMock());
-            Save();
+            SetupCategories();
+            SetupProducts();            
+            Database.EnsureCreated();
         }
 
-        private IEnumerable<ProductEntity> GetProductsMock()
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
-            var productModels = new List<ProductModel>
+            base.OnModelCreating(modelBuilder);           
+            modelBuilder
+                .Entity<CategoryEntity>()
+                .HasData(_categoryEntities);
+            modelBuilder                
+                .Entity<ProductEntity>()                
+                .HasData(_productModels);            
+        }
+
+        private void SetupProducts()
+        {
+            _productModels = new List<ProductEntity>
```

Comment:
Specify here categories for each product entity

Reply:
fixed